pallets/task/src/lib.rs

```diff
@@ -256,19 +262,42 @@ pub mod pallet {
 
 			// Emit a Task Removed Event.
 			Self::deposit_event(Event::TaskRemoved(signer, task_id));
-			
+
 			Ok(())
 		}
 	}
-	
+
+	#[pallet::hooks]
+	impl<T:Config> Hooks<T::BlockNumber> for Pallet<T> {
+		fn on_initialize(_n: T::BlockNumber) -> frame_support::weights::Weight {
+			let mut weight = 0;
+			let current_timestamp = T::Time::now();
+			let task_hashes : Vec<T::Hash> = Tasks::<T>::iter_keys().collect();
+			for th in task_hashes {
+				let task = Tasks::<T>::get(th);
+				if let Some(task) = task {
+					let deadline_duration = Duration::from_millis(task.deadline.saturated_into::<u64>());
+					if deadline_duration < current_timestamp {
+						if let Ok(()) = Self::delete_task(&task.initiator, &th) {
+							weight += 10_000;
```

Comment:
Looks good... is the weight estimate used for benchmarking. How do we know that 10_000 is the right weight for deleting tasks?